overlord/snapstate/handlers.go

```diff
@@ -3975,10 +3953,9 @@ func refreshedSnaps(reTask *state.Task) (snapNames []string, failed bool) {
 			laneSnaps = map[int]string{}
 		}
 		lanes := task.Lanes()
-		if len(lanes) != 1 {
-			// can't happen, really
-			continue
-		}
+		// Tasks can have multiple lanes, especially if a subset
+		// of them are transactional, the first lane of the snap task-set
+		// should always be a unique lane though.
```

Comment:
this comment is not quite true though if Flags.Transaction == "all-snap", there the snaps share a transaction as well, so lane[0] is not unique? is that code also broken with respect with that code? not really relevant for the PR, but the comment needs to accurate and somebody needs to look into fixing the code for that case if it's broken

Reply:
It made so much sense in my head when I wrote it - but you're absolutely correct. I've reworded this comment and added an XXX to point out this needs to be tested and maybe (most likely) fixed.